File: test-projects/test-java-project/src/main/java/com/github/vilmosnagy/elq/testproject/entities/Artist.java
package com.github.vilmosnagy.elq.testproject.entities;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

/**
 * @author Vilmos Nagy {@literal <vilmos.nagy@outlook.com>}
 */
@Entity
@Data
@EqualsAndHashCode
@Table(name = "artist")
public class Artist {

    @Id
    @Column(name="artistId")
    private int id;

    @Version
    @Column(name = "version")
    private Date version;

    @Column(name = "name")
    private String name;

    @OneToMany(mappedBy = "artist")
    private List<Album> albums;
}

File: test-projects/test-kotlin-project/src/main/kotlin/com/github/vilmosnagy/elq/testproject/entities/Artist.kt
package com.github.vilmosnagy.elq.testproject.entities

import java.util.*
import javax.persistence.*

/**
 * @author Vilmos Nagy {@literal <vilmos.nagy@outlook.com>}
 */
@Entity
@Table(name = "artist")
data class Artist(
        @Id @Column(name = "artistId")      override var id: Int,
        @Version @Column(name = "version")  override var version: Date,
        @Column(name = "name")              var name: String,
        @OneToMany(mappedBy = "artist")     var albums: MutableList<Album>
): IBaseEntity